File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        n=len(mat)
        count=0
        for i in range(n):
            count+=mat[i][i]
            count+=mat[i][n-i-1]
        if n%2==1:
            count-=mat[n//2][n//2]
        return count
        

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int row=mat.length;
        int total=0;
        for(int i=0;i<row;i++){
            total=total+mat[i][i];
            total=total+mat[i][row-i-1];
        }
        if(row%2!=0)
        total=total-mat[row/2][row/2];
        return total;
    }
    
}